src/buckets/components/createBucketForm/MeasurementSchemaSection.test.tsx

```diff
@@ -264,3 +266,60 @@ describe('test schema  update validity function', () => {
     doTest(null, true)
   })
 })
+
+describe('test csv conversion function (object->csv file)', () => {
+  it('should create a valid csv file', () => {
+    const csvString = toCsvString(oneColumns)
+
+    const lines = csvString.split('\n')
+    expect(lines[0]).toEqual('name,type,dataType')
+    expect(lines[1]).toEqual('time,timestamp,')
+    expect(lines[2]).toEqual('host,tag,')
+    expect(lines[3]).toEqual('service,tag,')
+    expect(lines[4]).toEqual('fsRead,field,float')
+    expect(lines[5]).toEqual('fsWrite,field,float')
+    expect(lines[6]).toEqual('timeSignature,tag,')
+    expect(lines[7]).toEqual('composer,tag,')
+    expect(lines[8]).toEqual('instrument,field,string')
+    expect(lines[9]).toEqual('one,field,string')
+
+    expect(lines.length).toEqual(10)
+  })
+})
+describe('test csv to array function (parsing)', () => {
```

Comment:
same request here for validating bad input

Reply:
it now checks that the column headers are correct.
then it parses it into an object, then the column data is checked; in the same exact way that the json data is checked; and there are already extensive tests for that.